exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionSenderRootExec.java

```diff
@@ -348,9 +348,12 @@ public void close() throws Exception {
     logger.debug("Partition sender stopping.");
     super.close();
     ok = false;
-    if (partitioner != null) {
-      updateAggregateStats();
-      partitioner.clear();
+
+    synchronized (this) {
```

Comment:
1. Should partitioner be volatile?
2. Should we check if partitioner is not null before synchronization as well (DCL)?